client/my-sites/plugins/plugin-automated-transfer/index.jsx

```diff
@@ -12,102 +12,135 @@ import wrapWithClickOutside from 'react-click-outside';
  */
 import { transferStates } from 'state/automated-transfer/constants';
 import { getSelectedSiteId } from 'state/ui/selectors';
-import { getAutomatedTransferStatus } from 'state/automated-transfer/selectors';
+import {
+	getAutomatedTransferStatus,
+	isAutomatedTransferTransferring,
+} from 'state/automated-transfer/selectors';
 import Notice from 'components/notice';
 import NoticeAction from 'components/notice/notice-action';
 
 class PluginAutomatedTransfer extends Component {
 
 	static propTypes = {
+		isTransferring: PropTypes.bool,
 		plugin: PropTypes.object,
-		status: PropTypes.string,
+		transferState: PropTypes.string,
 		translate: PropTypes.func,
-	}
+	};
 
 	static defaultProps = {
 		plugin: {
 			name: 'Plugin',
 		},
-	}
+	};
 
 	state = {
 		clickOutside: false,
+		shouldDisplay: true,
+		transferComplete: false,
+	};
+
+	componentWillMount() {
+		if ( transferStates.COMPLETE === this.props.transferState ) {
+			this.setState( { shouldDisplay: false } );
+		}
 	}
 
 	componentWillReceiveProps( nextProps ) {
-		if ( this.props.status !== nextProps.status ) {
+		const { COMPLETE, CONFLICTS } = transferStates;
+		const { isTransferring, transferState } = nextProps;
+
+		if ( COMPLETE === transferState ) {
+			this.setState( { transferComplete: true } );
+		}
+		if ( this.props.transferState !== transferState ) {
 			this.setState( { clickOutside: false } );
 		}
+		if (
+			this.state.shouldDisplay &&
+			! isTransferring &&
+			! includes( [ COMPLETE, CONFLICTS ], transferState )
+		) {
+			this.setState( { shouldDisplay: false } );
+		}
```

Comment:
we just have a single override on showing the progress bar, don't we? _if_ it's complete at mount time then hide it?

if that's true, what if we stored something like `hideProgressBar` as a boolean in component state and then allowed `render()` to compute the display each time, keeping the logic there?

```js
const shouldDisplay = isTransferring && ! this.state.hideDisplay
```

then there's no `componentWillReceiveProps` change. I'm not sure because we'd have to verify the logic of those statements, but maybe it's worth checking into. if possible, it would seem to reduce the complexity of the local component state and maybe limit the opportunity for bugs there.